polynote-spark/src/main/scala/polynote/kernel/lang/sql/SparkSqlInterpreter.scala

```diff
@@ -10,19 +10,24 @@ import fs2.Stream
 import fs2.concurrent.{Enqueue, Queue}
 import org.apache.spark.sql.catalyst.parser.{SqlBaseBaseVisitor, SqlBaseParser}
 import org.apache.spark.sql.catalyst.parser.SqlBaseParser.SingleStatementContext
+import org.apache.spark.sql.thief.SessionStateThief
 import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+import org.log4s.getLogger
 import polynote.kernel._
 import polynote.kernel.lang.LanguageInterpreter
 import polynote.kernel.util.{Publish, RuntimeSymbolTable}
 import polynote.messages.{CellID, ShortString, TinyList, TinyString}
 
 import scala.collection.mutable
+import scala.util.{Failure, Success}
 
 class SparkSqlInterpreter(val symbolTable: RuntimeSymbolTable) extends LanguageInterpreter[IO] {
 
-  import symbolTable.kernelContext, kernelContext.global
+  import symbolTable.kernelContext, kernelContext.global, kernelContext.implicits.executionContext
 
-  private implicit val contextShift: ContextShift[IO] = IO.contextShift(scala.concurrent.ExecutionContext.Implicits.global)
+  private val logger = getLogger
+
+  Thread.currentThread().setContextClassLoader(kernelContext.classLoader)
```

Comment:
It turns out you need to be really sure this is done. Since the interpreter is constructed in an IO, this can probably be removed after we make sure that all of the possible IO thread pools are doing this to all their threads.

Reply:
(Failure to set this causes things like `java.lang.ClassCastException: cannot assign instance of scala.collection.immutable.List$SerializationProxy to field org.apache.spark.rdd.RDD.org$apache$spark$rdd$RDD$$dependencies_ of type scala.collection.Seq in instance of org.apache.spark.rdd.MapPartitionsRDD`)

Reply:
For safety until the rest of the thread pools are handled should we throw this into the other interpreters as well?

Reply:
I think it only becomes an issue if you do a spark operation outside of an `IO`.